### PR TITLE
SC-1856: Support docker machine

### PR DESCRIPTION
**Pull request by zyuzka**
Ticket: https://spryker.atlassian.net/browse/SC-1856
RG: https://release.spryker.com/release-groups/view/2385

### CODE REVIEW

File: bin/docker-machine/parallels.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+. ./abstract-docker-machine.sh
+
+../require.sh prlctl
+popd > /dev/null
+
+function exportDockerMachineConfiguration()
+{
+    export DOCKER_MACHINE_DRIVER=parallels
+    export DOCKER_MACHINE_NAME=${SPRYKER_DOCKER_PREFIX}-${SPRYKER_DOCKER_TAG}-${DOCKER_MACHINE_DRIVER}-machine
+    export DOCKER_MACHINE_CPU=2
```
Comment (alexanderM91) [inline bin/docker-machine/parallels.sh]: Configuration should be adjustable using docker sdk declaration.

File: bin/docker-machine/virtualbox.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+. ./abstract-docker-machine.sh
+
+../require.sh vboxmanage
+popd > /dev/null
+
+function exportDockerMachineConfiguration()
+{
+    export DOCKER_MACHINE_DRIVER=virtualbox
+    export DOCKER_MACHINE_NAME=${SPRYKER_DOCKER_PREFIX}-${SPRYKER_DOCKER_TAG}-${DOCKER_MACHINE_DRIVER}-machine
+    export DOCKER_MACHINE_CPU=2
```
Comment (alexanderM91) [inline bin/docker-machine/virtualbox.sh]: the same as above

File: bin/docker-machine/virtualbox.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+popd > /dev/null
+
+function getDockerMachineArguments()
+{
+    local dockerMachineDriver=virtualbox
+    local dockerMachineName=${SPRYKER_DOCKER_PREFIX}-${SPRYKER_DOCKER_TAG}-${dockerMachineDriver}-machine
+
+    echo "${DOCKER_MACHINE_NAME:-${dockerMachineName}} \
+        --driver=${dockerMachineDriver} \
+        --parallels-cpu-count=${DOCKER_MACHINE_CPU:-2} \
```
Comment (sprymiker) [inline bin/docker-machine/virtualbox.sh]: parallels? for VB?

File: bin/docker-machine/abstract-docker-machine.sh [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+set -e
+
+pushd ${BASH_SOURCE%/*} > /dev/null
+. ../constants.sh
+. ../console.sh
+
+../require.sh docker-machine
+popd > /dev/null
+
+function isDockerMachineExist()
```
Comment (sprymiker) [inline bin/docker-machine/abstract-docker-machine.sh]: doesDockerMachineExist

File: bin/docker-machine/abstract-docker-machine.sh [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    local dockerMachineName=${1}
+    local dockerMachineDriver=${2}
+    local dockerMachineCpu=${3}
+    local dockerMachineMemory=${4}
+    local dockerMachineDiskSize=${5}
+    local dockerMachineSharedFolder=${6}
+
+    echo -e "${INFO}Creating ${LGRAY}"${dockerMachineName}"${INFO} machine... ${NC}"
+
+    docker-machine create ${dockerMachineDriver} \
```
Comment (sprymiker) [inline bin/docker-machine/abstract-docker-machine.sh]: Why would not just use `$@` ?

File: generator/src/templates/deploy.bash.twig [hunk 6]
```diff
@@ -726,6 +742,7 @@ case ${command} in
         docker system prune -a
         docker image prune
         docker volume prune
+        deleteDockerMachine $(getDockerMachineArguments)
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig]: Do we need this for `clean` command?

Comment (zyuzka) [reply generator/src/templates/deploy.bash.twig]: I don't think so

File: bin/docker-machine/virtualbox.sh (hunk 3 above)
Comment (zyuzka) [reply bin/docker-machine/virtualbox.sh]: Fixed

File: bin/docker-machine/abstract-docker-machine.sh (hunk 4 above)
Comment (zyuzka) [reply bin/docker-machine/abstract-docker-machine.sh]: Fixed

File: bin/docker-machine/abstract-docker-machine.sh (hunk 5 above)
Comment (zyuzka) [reply bin/docker-machine/abstract-docker-machine.sh]: fixed

**Pull request closed without merging**